astropy/coordinates/earth.py

```diff
@@ -174,6 +176,125 @@ def from_geodetic(cls, lon, lat, height=0., ellipsoid=None):
+            cannot (i.e., an internet connection is not available), it will fall
+            back on the list included with astropy (which is a limited and dated
+            set of sites).  If you think a site should be added, issue a pull
+            request to the
+            `astropy-data repository <https://github.com/astropy/astropy-data>`_ .
+
+        Parameters
+        ----------
+        site_name : str
+            Name of the observatory (case-insensitive).
+
+        Returns
+        -------
+        site : This class (a `~astropy.coordinates.EarthLocation` or subclass)
+            The location of the observatory.
+
+        See Also
+        --------
+        get_site_names : the list of sites that this function can access
+        """
+        el = cls._get_site_registry().get_site(site_name)
+
+        if cls is el.__class__:
+            return el
+        else:
+            neweel = cls.from_geodetic(*el.to_geodetic())
+            newel.info.name = el.info.name
+            return newel
+
+    @classmethod
+    def get_site_names(cls):
+        """
+        Get list of names of observatories for use with
+        `~astropy.coordinates.EarthLocation.of_site`.
+
+        .. note::
+            When this function is called, it will first attempt to
+            download site information from the astropy data server.  If it
+            cannot (i.e., an internet connection is not available), it will fall
+            back on the list included with astropy (which is a limited and dated
+            set of sites).  If you think a site should be added, issue a pull
+            request to the
+            `astropy-data repository <https://github.com/astropy/astropy-data>`_ .
+
+
+        Returns
+        -------
+        names : list of str
+            List of valid observatory names
+
+        See Also
+        --------
+        of_site : Gets the actual location object for one of the sites names
+                  this returns.
+        """
+        return cls._get_site_registry().get_site_names()
+
+    @classmethod
+    def _get_site_registry(cls, force_download=False, force_builtin=False):
+        """
+        Gets the site registry.  The first time this either downloads or loads
+        from the data file packaged with astropy.  Subsequent calls will use the
+        cached version unless explicitly overridden.
+
+        Parameters
+        ----------
+        force_download : bool or str
+            If not False, force replacement of the cached registry with a
+            downloaded version. If a str, that will be used as the URL to
+            download from (if just True, the default URL will be used).
+        force_builtin : bool
+            If True, load from the data file bundled with astropy and set the
+            cache to that.
+
+        returns
+        -------
+        reg : astropy.coordinates.sites.SiteRegistry
+        """
+        # need to import inside function to avoid circular dependencies
```

Comment:
I learned from @embray that one trick to avoid this is to put the import at the very end of the file. I *think* it should work here too.